src/metabase/integrations/slack.clj

```diff
@@ -7,13 +7,7 @@
 
 
 ;; Define a setting which captures our Slack api token
-(defsetting slack-token "Slack API bearer token obtained from https://api.slack.com/web#authentication" nil
-  :setter (fn [new-value]
-            (setting/set* :slack-token new-value)
-            (require 'metabase.metabot)
-            ((ns-resolve 'metabase.metabot (if (seq new-value)
-                                              'start-metabot!
-                                              'stop-metabot!)))))
+(defsetting slack-token "Slack API bearer token obtained from https://api.slack.com/web#authentication" nil)
```

Comment:
got it.  couple reasons to go with the events system instead of the defsetting `:setter` option are that 1) i'd prefer to just use the events stuff if possible and be consistent wherever possible, and 2) in this specific case i think it's cleaner to use the events stuff because it removes the direct coupling between the setting of the value and the metabot stuff.  there is no real reason why the 2 pieces of code need to know about each other and using the events system they don't.